records.go

```diff
@@ -98,13 +98,12 @@ func (dht *IpfsDHT) getPublicKeyFromNode(ctx context.Context, p peer.ID) (ci.Pub
 
 	// Get the key from the node itself
 	pkkey := routing.KeyForPublicKey(p)
-	pmes, err := dht.getValueSingle(ctx, p, pkkey)
+	record, _, err := dht.protoMessenger.GetValue(ctx, p, pkkey)
```

Comment:
This is _slightly_ more work since we're bothering to decode to the closer peers in the protobuf but that doesn't really seem like a big deal.